Source/WebKit/NetworkProcess/cocoa/NetworkTaskCocoa.mm

```diff
@@ -127,15 +141,35 @@ static bool shouldCapCookieExpiryForThirdPartyIPAddress(const WebCore::IPAddress
     return quirk->value == requestDomain;
 }
 
-void NetworkTaskCocoa::applyCookiePolicyForThirdPartyCloaking(const WebCore::ResourceRequest& request)
+void NetworkTaskCocoa::setCookieTransform(const WebCore::ResourceRequest& request)
 {
+#if HAVE(ALLOW_ONLY_PARTITIONED_COOKIES)
+    bool needsQuirk = Quirks::needsPartitionedCookies(request);
+
+    if (!shouldApplyCookiePolicyForThirdPartyCloaking() && !needsQuirk)
+        return;
+
+    CheckedPtr networkStorageSession = m_networkSession->networkStorageSession();
+    if (!networkStorageSession) {
+        task()._cookieTransformCallback = nil;
+        return;
+    }
+
+    if (!networkStorageSession->isOptInCookiePartitioningEnabled()) {
+        task()._cookieTransformCallback = nil;
+        return;
+    }
+
+    auto cookiePartition = networkStorageSession->cookiePartitionIdentifier(request);
+#else
```

Comment:
Is this correct? 
Before this change, if `shouldApplyCookiePolicyForThirdPartyCloaking` is true and `networkStorageSession->isOptInCookiePartitioningEnabled()` is false, we will set `_cookieTransformCallback`, if request is first-party; with this change it seems we don't set it (due to the early return at line 160).

It might be clearer to not mix these two cases. My understanding is one will only be applied to first-party request and the other is only applied to third-party request, so maybe we could have`setCookieTransformForFirstPartyRequest` and `setCookieTransformForThirdPartyRequest`, and ensure only one setting will succeed.

Reply:
Good catch, that check should be `needsQuirk && !networkStorageSession->isOptInCookiePartitioningEnabled()`. We could split these into two functions, but I was concerned about them conflicting in the future. I agree they should be mutually exclusive right now.